rust/origen/src/generator/ast.rs

```diff
@@ -299,7 +303,7 @@ impl AST {
         self.nodes.push(node);
     }
 
-    pub fn process(&self, process_fn: &dyn Fn(&Node) -> Node) -> Node {
+    pub fn process(&self, process_fn: &mut dyn FnMut(&Node) -> Node) -> Node {
```

Comment:
What's the purpose/effect of this change?

Reply:
Does it allow the closure to mutate local vars?

Reply:
Yes. I added a local state to the generators that it can use for any metadata it thinks it needs while either intercepting or processing. In order to do that though, I had to give the `test_manager.process` function a mutable reference. [Here's a place where I use this.](https://github.com/Origen-SDK/o2/blob/tester_prototype/rust/origen/src/testers/mod.rs#L80-L84)

Probably could come up with something else if need be, but I don't think there's harm in giving it a mutable reference since the tester will only have its own processors which can mutate its state.

Reply:
To go into a bit more details: the generators (on both the Python and Rust sides) are instantiated when they are targeted and will maintain their state until `tester.reset` is called, which clears all the targets.

Reply:
OK, fair enough, thanks